packages/core/src/browser/shell/application-shell.ts

```diff
@@ -1618,12 +1618,32 @@ export class ApplicationShell extends Widget {
         return [...this.tracker.widgets];
     }
 
-    canToggleMaximized(): boolean {
-        const area = this.currentWidget && this.getAreaFor(this.currentWidget);
-        return area === 'main' || area === 'bottom';
+    canToggleMaximized(widget: TabBar<Widget>): boolean {
+        if (widget instanceof Widget) {
+            const area = this.getAreaFor(widget);
+            return area === 'main' || area === 'bottom';
+        }
+        return false;
+    }
+
+    toggleMaximized(options?: { currentTitle: Title<Widget> | undefined, widget: TabBar<Widget> }): void {
+        if (options) {
+            const { currentTitle, widget } = options;
+            if (widget instanceof Widget && currentTitle instanceof Title) {
+                const title = widget.titles.find(t => t.caption === currentTitle.caption);
+                const area = this.getAreaPanelFor(title ? title.owner : this.currentWidget);
+                if (area instanceof TheiaDockPanel && (area === this.mainPanel || area === this.bottomPanel)) {
+                    area.toggleMaximized();
+                }
+            } else {
+                this.oldToggleMaximized();
+            }
+        } else {
+            this.oldToggleMaximized();
+        }
     }
 
-    toggleMaximized(): void {
+    private oldToggleMaximized(): void {
```

Comment:
@Anasshahidd21 why do we need `oldToggleMaximized`?

Reply:
updated.